### PR TITLE
[SYSTEMML-1015] Add SystemML Roadmap to website

### PR DESCRIPTION
**Pull request by dlesaca**
- Added roadmap to site architecture

### CODE REVIEW

**Comment by dlesaca** [comment]
Yo, check it! @lresende @deroneriksson @dusenberrymw

**Comment by dusenberrymw** [comment]
@dlesaca Merged.  I believe a few of the commits in this PR were actually existing ones that happened to sneak in due to the history rewrite.  Likely, this happened due to starting this work prior to the git changes.  No worries though, just please double check the latest master right now to ensure that everything is in that should be there.  Thanks!

**Comment by dusenberrymw** [comment]
Quick update: A more thorough analysis of the previous attempt to merge this PR revealed that it had inadvertently overwritten previous updates made since this PR was submitted.  Therefore, I reverted that commit, and re-merged this PR.  Additionally, I pushed one fix based on the filename change from "get-involved.html" -> "banner_social.html" that affected other work that was done after this PR was submitted.  The result is that the current master is now correct, aligns correctly with the changes made in this PR, and builds.

tl;dr: looks good now 👍
